<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>D405 Farm Recorder (PyQt5)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_root">
    <!-- ======= 상단 컨트롤 ======= -->
    <item>
     <layout class="QGridLayout" name="grid_controls">
      <item row="0" column="0">
       <widget class="QPushButton" name="btnStart">
        <property name="text"><string>Start</string></property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btnStop">
        <property name="text"><string>Stop</string></property>
        <property name="enabled"><bool>false</bool></property>
       </widget>
      </item>

      <item row="0" column="2">
       <widget class="QLabel" name="lblOutDir">
        <property name="text"><string>Output Dir</string></property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLineEdit" name="editOutDir">
        <property name="placeholderText"><string>save</string></property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btnBrowse">
        <property name="text"><string>Browse...</string></property>
       </widget>
      </item>

      <item row="1" column="2">
       <widget class="QLabel" name="lblRes">
        <property name="text"><string>Resolution</string></property>
       </widget>
      </item>
      <item row="1" column="3">
        <widget class="QComboBox" name="comboRes">
         <item><property name="text"><string>640x480</string></property></item>
         <item><property name="text"><string>848x480</string></property></item>
         <item><property name="text"><string>1280x720</string></property></item>
        </widget>
      </item>

      <item row="1" column="4">
       <widget class="QLabel" name="lblFps">
        <property name="text"><string>FPS</string></property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QComboBox" name="comboFps">
        <item><property name="text"><string>15</string></property></item>
        <item><property name="text"><string>30</string></property></item>
       </widget>
      </item>

      <item row="2" column="2">
       <widget class="QCheckBox" name="chkSaveVideos">
        <property name="text"><string>Save MP4</string></property>
        <property name="checked"><bool>false</bool></property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="chkSaveBag">
        <property name="text"><string>Save .bag</string></property>
        <property name="checked"><bool>false</bool></property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QCheckBox" name="chkSaveDepthRaw">
        <property name="text"><string>Save Depth (.npy)</string></property>
        <property name="checked"><bool>true</bool></property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QCheckBox" name="chkSaveFrames">
        <property name="text"><string>Save Frames</string></property>
        <property name="checked"><bool>true</bool></property>
       </widget>
      </item>
      <item row="2" column="5">
       <widget class="QLabel" name="lblFrameStep">
        <property name="text"><string>Frame Step</string></property>
       </widget>
      </item>
      <item row="2" column="6">
       <widget class="QSpinBox" name="spinFrameStep">
        <property name="minimum"><number>1</number></property>
        <property name="maximum"><number>120</number></property>
        <property name="value"><number>1</number></property>
       </widget>
      </item>

      <item row="3" column="0" colspan="7">
       <widget class="QLabel" name="lblStatus">
        <property name="text"><string>Status: Idle</string></property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- ======= 미리보기: Depth + Color ======= -->
    <item>
     <layout class="QHBoxLayout" name="hbox_previews">
      <item>
       <widget class="QLabel" name="viewDepth">
        <property name="minimumSize"><size><width>320</width><height>240</height></size></property>
        <property name="frameShape"><enum>QFrame::Box</enum></property>
        <property name="text"><string>Depth (colorized)</string></property>
        <property name="alignment"><set>Qt::AlignCenter</set></property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="viewColor">
        <property name="minimumSize"><size><width>320</width><height>240</height></size></property>
        <property name="frameShape"><enum>QFrame::Box</enum></property>
        <property name="text"><string>Color (RGB)</string></property>
        <property name="alignment"><set>Qt::AlignCenter</set></property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- ======= NPY 뷰어 자리 ======= -->
    <item>
     <widget class="QGroupBox" name="groupViewer">
      <property name="title"><string>NPY Viewer</string></property>
      <layout class="QVBoxLayout" name="vbox_viewer">
       <item>
        <widget class="QWidget" name="viewerNpyContainer">
         <property name="minimumSize">
          <size><width>100</width><height>200</height></size>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
